### DISCUSSION TITLE
quasar command with npm run

### DISCUSSION THREAD

**Discussion by minyoung90** [quasarframework/quasar#15080] [Show and Tell]
quasar has a default command for "quasar build" as "npm run build"
what I want to do is to make command "npm run serve"

So, what I did was to set like below

```javascript
// package.json
{
    "scripts": {
     // ....
        "build": "quasar build",
        "serve": "quasar serve ./dist/spa",
     },
}
```

command "quasar serve ./dist/spa" is fine when it is executed independently but, if i run "npm run serve" I got an error message below.

```
> quasar-ckeditor-webpack@0.0.1 serve
> quasar serve ./dist/spa

 App • Looking for Quasar App Extension "serve" command "./dist/spa"
 App • ⚠️  Quasar App Extension "serve" is missing...
```

What is wrong? Please help me!

**Comment by tofi86** [comment]
I have the same issue. 'serve' is provided by `@quasar/cli` whereas the other commands are provided by `@quasar/app-webpack` or `@quasar/app-vite`. You can also notice which packages get loaded by Quasar right under the Quasar ASCI art logo. And when running with `npm`, you can see that only `@quasar/app-webpack` gets loaded. Therefore it cannot find 'serve'. That said, I haven't found a solution for this yet... Did you @minyoung90 ?

**Comment by AlexDaniel** [comment]
Huh, I have the same `"serve" is missing` problem, as if the command is just not there. I'm able to serve the production build by using `vite` directly:
```bash
pnpm vite serve dist/spa/
```

**Comment by yusufkandemir** [comment]
The `serve` command is provided by the global CLI (`@quasar/cli`). So, you have two solutions:
1. Use global execution(npx, pnpm dlx, etc.): e.g. `"serve": "npx @quasar/cli serve ./dist/spa",`
2. Install `@quasar/cli` locally: e.g. `npm i --save-dev @quasar/cli`
	- If using pnpm, it resolves things a bit differently, so the package script need to be adjusted as well: `"serve": "./node_modules/@quasar/cli/bin/quasar.js serve ./dist/spa",`

**Comment by tofi86** [comment]
@yusufkandemir Your first solution didn't work for me either:

```
npx quasar serve ./dist/spa
 App • Looking for Quasar App Extension "serve" command "--help"
 App • ⚠️  Quasar App Extension "serve" is missing...

 .d88888b.
d88P" "Y88b
888     888
888     888 888  888  8888b.  .d8888b   8888b.  888d888
888     888 888  888     "88b 88K          "88b 888P"
888 Y8b 888 888  888 .d888888 "Y8888b. .d888888 888
Y88b.Y8b88P Y88b 888 888  888      X88 888  888 888
 "Y888888"   "Y88888 "Y888888  88888P' "Y888888 888
       Y8b

  Running @quasar/app-vite v1.6.2

```

Your second solution did work though. Thanks!

**Reply by yusufkandemir** [reply]
Oops, it was supposed to be `npx @quasar/cli serve ./dist/spa`, updated my message. But, I prefer and recommend the second solution.
